src/Libraries/Microsoft.Extensions.AI/Utilities/AIJsonUtilities.Sse.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization.Metadata;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+using Microsoft.Shared.Pools;
+
+#pragma warning disable SA1114 // Parameter list should follow declaration
+
+namespace Microsoft.Extensions.AI;
+
+public static partial class AIJsonUtilities
+{
+    private static readonly byte[] _sseEventFieldPrefix = "event: "u8.ToArray();
+    private static readonly byte[] _sseDataFieldPrefix = "data: "u8.ToArray();
+    private static readonly byte[] _sseIdFieldPrefix = "id: "u8.ToArray();
+    private static readonly byte[] _sseLineBreak = Encoding.UTF8.GetBytes(Environment.NewLine);
```

Comment:
Using the OS-specific line break felt appropriate given that all variants are supported by the spec.

Reply:
Most implementations I've seen always use '\n' regardless of OS.